apps/designer/app/routes/designer/$id.tsx

```diff
@@ -88,11 +89,12 @@ export const action: ActionFunction = async ({ request, params }) => {
 const DesignerRoute = () => {
   const actionData = useActionData();
   const data = useLoaderData<Data | Error>();
+
   if ("errors" in data) {
     return <ErrorMessage message={data.errors} />;
   }
   if (actionData && "errors" in actionData) {
-    return <ErrorMessage message={actionData.errors} />;
+    toast.error(actionData.errors, { id: "upload-error" });
```

Comment:
The id makes sure only one of them appears

Reply:
hmm, but if there are multiple once, shouldn't they all be somehow visible to the user? should they potentially get merged into one toast?

Reply:
The issue is that it was running the same error twice

Reply:
hmm, why though?

Reply:
All very valid questions ahaha

It may be because it should be in a useEffect, let me check

Reply:
100%, you also don't need that in SSR and useEffect won't run there

Reply:
Adding a useEffect did fix it